### PR TITLE
Remove Discourse admin API key and user.

### PR DESCRIPTION
**Pull request by Beanow**
This removes all usage of and reference to the admin API key and username. Instead relying on anonymous access of the Discourse API.

This enables anyone to deploy an instance with discourse support, and is much safer, since the admin API key isn't used for this purpose anymore. Once merged I would encourage revoking any admin API keys used in the past.

The only notable remaining reference of the discourse username is in the project file.
Which goes from `0.3.0` to `0.3.1` in a backwards-compatible way here, simply ignoring the username if present. For #1426 I'm expecting a `0.4.0` version, so this is to prevent having to change project files twice.

Test plan: updated the snapshots to their latest anonymous versions. Ran yarn test and anonymous discourse loading from CLI numerous times.

### CODE REVIEW

File: src/core/project.js [hunk 1]
```diff
@@ -42,7 +44,7 @@ export function projectToJSON(p: Project): ProjectJSON {
 }
 
 export function projectFromJSON(j: ProjectJSON): Project {
-  return fromCompat(COMPAT_INFO, j);
+  return fromCompat(COMPAT_INFO, j, upgrades);
```
Comment (decentralion) [inline src/core/project.js]: Nice! I think this is the first use of the `upgrades` API. @wchargin and I have gone back and forth a bit on the quality of this API, I'd be curious to hear what you thought of this interface.

**Comment by decentralion** [comment]
Question: are any of the snapshot updates actually a result of the changes in this PR, or are they just drift on upstream? (The snapshot tests are never consistent from snapshot to snapshot because, among other things, the view count is returned by the server.)

In general I try to separate irrelevant / noisy snapshot updates from the PRs that touch DIscourse, for easier and clearer review. My ideal pattern would be:
- PR with pre-snapshot update (trivial approve)
- PR with the code changes, with another snapshot update included (but this one will only reveal differences due to the code change, as opposed to change on the upstream)

**Comment by Beanow** [comment]
@decentralion
> Question: are any of the snapshot updates actually a result of the changes in this PR

There are actual changes. Because the requesting user (anonymous) has fewer privileges. It removes some "can do this action: true" style fields, "have you read this yet" and data like user creation dates. None of which we actively used internally, but they're different between using and not using the API key.

**Comment by decentralion** [comment]
Great. I never wanted that user-specific data to begin with.

**Pull request merged**